### ISSUE TITLE
instagram banned my account

### ISSUE THREAD

**Issue by helallao** [adw0rd/instagrapi#581]
I made a meme page generator, it just copies posts from my meme pages list but instagram banned my account and deleted all my posts. 

![Screenshot_20220218-152603_Instagram](https://user-images.githubusercontent.com/78656003/154682725-d88679ab-ef2e-490a-bf5d-c593c2e46728.jpg)

I added phone number but my script stopped working for this account

Doesn't this repository use private api?
Why i got banned?
How can i pass this?

code:
```python import random, instagrapi, time, os
from datetime import datetime



cl = instagrapi.Client()
cl.login('username', 'password')




history = []
copy_list = ['dugunteroru', 'editah4z', 'tassakshitpost', 'boreklipogaca', 'medraxhumor', 'jolithex', 'palanin_mekani', 'igcoplugu', 'the_.green._goblin', 'parttimeejderyaavcisi', 'wiptieyerli']



def clear_files():
    for i in os.listdir('./myfiles/'):
        os.remove(f'./myfiles/{i}')




def send_follow(id):
    for i in cl_2.user_following(cl_2.user_id, 10):
        try:
            cl_2.user_unfollow(i)
            
        except:
            break


    comments = cl_2.media_comments(id)
    
    for i in comments[:10]:
        try:
            id = cl_2.user_id_from_username(i.user.username)
            
            cl_2.user_follow(id)
        
        except:
            break
        
        #follow users from the post comments




def send_post():
    while True:
        page_name = random.choice(copy_list)
        copy_page = cl_2.user_id_from_username(page_name)
        
        post = random.choice(cl_2.user_medias(copy_page, 10))
        
        if post.code not in history:
            history.append(post.code)
            break
        
        #choose a random post from page, and save code to history
    
    
    
    send_follow(post.id)
    
    
    
    if post.media_type == 1:
        try:
            photo = cl_2.photo_download(cl_2.media_pk_from_code(post.code), './myfiles')
            
            cl_2.photo_upload(photo, post.caption_text)
            
            clear_files()
        
        
        except:
            pass
    
    
    
    elif post.media_type == 2 and post.product_type == 'feed':
        try:
            video = cl_2.video_download(cl_2.media_pk_from_code(post.code), './myfiles')
            
            cl_2.video_upload(video, post.caption_text)
            
            clear_files()
        
        
        except:
            pass
    
    
    
    
    elif post.media_type == 2 and post.product_type == 'igtv':
        try:
            igtv = cl_2.igtv_download(cl_2.media_pk_from_code(post.code), './myfiles')
            
            cl_2.igtv_upload(igtv, post.caption_text)
            
            clear_files()
        
        
        except:
            pass
    
    
    
    
    elif post.media_type == 2 and post.product_type == 'clips':
        try:
            clip = cl_2.clip_download(cl_2.media_pk_from_code(post.code), './myfiles')
            
            cl_2.clip_upload(clip, post.caption_text)
            
            clear_files()
        
        
        except:
            pass
    
    
    
    
    elif post.media_type == 8:
        try:
            album = cl_2.album_download(cl_2.media_pk_from_code(post.code), './myfiles')
            
            cl_2.album_upload(album, post.caption_text)
            
            clear_files()
        
        
        except:
            pass





def run():
    while True:
        time.sleep(60 * random.randint(20, 45))
        #sleep 20 ~ 45 minutes
        
        send_post()



run()
```my meme pages list but instagram banned my account and deleted all my posts. 

Doesn't this repository use private api?
Why i got banned?
How can i pass this?

code:
```python import random, instagrapi, time, os
from datetime import datetime



cl = instagrapi.Client()
cl.login('username', 'password')




history = []
copy_list = ['dugunteroru', 'editah4z', 'tassakshitpost', 'boreklipogaca', 'medraxhumor', 'jolithex', 'palanin_mekani', 'igcoplugu', 'the_.green._goblin', 'parttimeejderyaavcisi', 'wiptieyerli']



def clear_files():
    for i in os.listdir('./myfiles/'):
        os.remove(f'./myfiles/{i}')




def send_follow(id):
    for i in cl_2.user_following(cl_2.user_id, 10):
        try:
            cl_2.user_unfollow(i)
            
        except:
            break


    comments = cl_2.media_comments(id)
    
    for i in comments[:10]:
        try:
            id = cl_2.user_id_from_username(i.user.username)
            
            cl_2.user_follow(id)
        
        except:
            break
        
        #follow users from the post comments




def send_post():
    while True:
        page_name = random.choice(copy_list)
        copy_page = cl_2.user_id_from_username(page_name)
        
        post = random.choice(cl_2.user_medias(copy_page, 10))
        
        if post.code not in history:
            history.append(post.code)
            break
        
        #choose a random post from page, and save code to history
    
    
    
    send_follow(post.id)
    
    
    
    if post.media_type == 1:
        try:
            photo = cl_2.photo_download(cl_2.media_pk_from_code(post.code), './myfiles')
            
            cl_2.photo_upload(photo, post.caption_text)
            
            clear_files()
        
        
        except:
            pass
    
    
    
    elif post.media_type == 2 and post.product_type == 'feed':
        try:
            video = cl_2.video_download(cl_2.media_pk_from_code(post.code), './myfiles')
            
            cl_2.video_upload(video, post.caption_text)
            
            clear_files()
        
        
        except:
            pass
    
    
    
    
    elif post.media_type == 2 and post.product_type == 'igtv':
        try:
            igtv = cl_2.igtv_download(cl_2.media_pk_from_code(post.code), './myfiles')
            
            cl_2.igtv_upload(igtv, post.caption_text)
            
            clear_files()
        
        
        except:
            pass
    
    
    
    
    elif post.media_type == 2 and post.product_type == 'clips':
        try:
            clip = cl_2.clip_download(cl_2.media_pk_from_code(post.code), './myfiles')
            
            cl_2.clip_upload(clip, post.caption_text)
            
            clear_files()
        
        
        except:
            pass
    
    
    
    
    elif post.media_type == 8:
        try:
            album = cl_2.album_download(cl_2.media_pk_from_code(post.code), './myfiles')
            
            cl_2.album_upload(album, post.caption_text)
            
            clear_files()
        
        
        except:
            pass





def run():
    while True:
        time.sleep(60 * random.randint(20, 45))
        #sleep 20 ~ 45 minutes
        
        send_post()



run()
```

**Comment by Jmallone** [comment]
If you only post with your personal account and nothing else, I recommend using the [official API](https://developers.facebook.com/docs/instagram-api/), differences:
- You will not be able to post Stories;
- Must be a [Business account](https://help.instagram.com/502981923235522);
- You have to renew the Token every 3 months, more or less.

<hr>

 I have an account that generates images and I use the official `API` to post to the feed, and the **instagrapi** to post to Stories.
 And it's been 8 months since everything is going well, maybe it's an alternative for you.

**Comment by savosta1** [comment]
tell me today this method will help unlock your instagram account?

**Comment by rehman-00001** [comment]
but using the official API requires facebook to verify/review the app right?

![Screenshot 2023-04-08 at 6 34 51 PM](https://user-images.githubusercontent.com/49582824/230722724-99193d0f-c9fe-483d-961a-7d6f6e708b78.png)

@Jmallone , how did you get past this step?

**Comment by diezo** [comment]
This could possibly be because instagrapi doesn't have an auto session management feature and you've logged in a lot of times.

You should try using [ensta](https://github.com/diezo/Ensta). While it uses the web api, it stores your previous session locally.

**Comment by Richoffjuggzz** [comment]
> To get your account unbanned, temporarily give insta your phone number (you can remove it after). And use `client.load_settings` as shown [here](https://adw0rd.github.io/instagrapi/usage-guide/interactions.html#settings)

can you send me that the link isnt working anymore

**Comment by subzeroid** [comment]
This is not actionable as a library bug. Instagram bans/locks are account enforcement decisions triggered by account history, repeated uploads/follows/comments, device state, proxy/IP reputation, and platform policy signals. `instagrapi` cannot unban or bypass that state.

The current docs for reducing account-risk are here:

- Best practices: https://subzeroid.github.io/instagrapi/usage-guide/best-practices
- Session reuse/settings: https://github.com/subzeroid/instagrapi#session-persistence
- Challenge handling: https://subzeroid.github.io/instagrapi/usage-guide/challenge_resolver

The short version is: use stable sessions (`dump_settings` / `load_settings`), avoid repeated fresh logins, keep proxy/device identity stable, slow down write-heavy automation, and resolve account-review states manually in the official app when Instagram requires it.

Closing this as account-support / not a reproducible instagrapi defect.

**Comment by subzeroid** [comment]
Closed as account-support, not a reproducible library defect.
